.circleci/test-deploy.yml

```diff
@@ -0,0 +1,147 @@
+
+# ----------------
+# Orbs declaration
+# ----------------
+
+orbs:
+  noos-ci: noosenergy/noos-ci@dev:<<pipeline.git.revision>>
+  orb-tools: circleci/orb-tools@12.0.4
+
+release-filters: &release-filters
+  tags:
+    # ONLY git tag compliant format: e.g. `major-release-1.0.0`
+    only: /^(major|minor|patch)-release-v\[0-9]+(\.[0-9]+)*(-alpha\.[0-9]+)?/
+  branches:
+    ignore: /.*/
+
+# --------------
+# Pipeline tasks
+# --------------
+
+jobs:
+
+  integration-test_python-lint-package:
+    executor:
+      name: noos-ci/default
+      tag: "3.11.4"
+    steps:
+      - checkout
+      - run:
+          name: Copy fixtures files over
+          command: cp -R ./fixtures/python/* .
+      - noos-ci/python-lint-package:
+          pkg-manager: pipenv
+          install-args: "--dev"
+          configure-git: false
+          src-directory: "./src"
+
+  integration-test_python-test-package:
+    executor:
+      name: noos-ci/default
+      tag: "3.11.4"
+    steps:
+      - checkout
+      - run:
+          name: Copy fixtures files over
+          command: cp -R ./fixtures/python/* .
+      - noos-ci/python-test-package:
+          pkg-manager: pipenv
+          install-args: "--dev"
+          configure-git: false
+          tests-directory: "./src/tests"
+
+  integration-test_python-build-wheel:
+    executor:
+      name: noos-ci/default
+      tag: "3.11.4"
+    steps:
+      - checkout
+      - run:
+          name: Copy fixtures files over
+          command: cp -R ./fixtures/python/* .
+      - noos-ci/python-build-wheel:
+          pkg-manager: pipenv
+          install-args: "--dev"
+          configure-git: false
+
+  integration-test_helm-lint-chart:
+    executor:
+      name: noos-ci/default
+      tag: "3.11.4"
+    steps:
+      - checkout
+      - noos-ci/helm-lint-chart:
+          chart-directory: ./fixtures/helm
+
+  integration-test_helm-build-chart:
+    executor:
+      name: noos-ci/default
+      tag: "3.11.4"
+    steps:
+      - checkout
+      - noos-ci/helm-build-chart:
+          registry-provider: aws
+          chart-directory: ./fixtures/helm
+          chart-name: noos-prod/noos-test
+          chart-tag: 0.0.1
+
+  integration-test_docker-build-image:
+    executor:
+      name: noos-ci/default
+      tag: "3.11.4"
+    steps:
+      - checkout
+      - noos-ci/docker-build-image:
+          registry-provider: docker
+          configure-git: false
+          image-context: ./fixtures/docker
+          image-name: busybox
+          image-tag: ${CIRCLE_SHA1}
+
+  integration-test_terraform-run-plan:
+    executor:
+      name: noos-ci/default
+      tag: "3.11.4"
+    steps:
+      - checkout
+      - noos-ci/terraform-update-variable:
+          workspace: noos-prod
+          variable-name: noos_ci_integration_test
+          variable-value: ${CIRCLE_SHA1}
+      - noos-ci/terraform-run-plan:
+          workspace: noos-prod
+          plan-message: ${CIRCLE_BUILD_URL}
+
+
+workflows:
+  test-publish:
```

Comment:
My apologies if this wasn't clear ; but as written, this workflow only includes testing job (no publishing action here).